core/dbt/include/global_project/macros/adapters/apply_grants.sql

```diff
@@ -34,54 +34,76 @@
 {% endmacro %}
 
 {%- macro default__get_grant_sql(relation, grant_config) -%}
-    {%- for privilege in grant_config.keys() -%}
```

Comment:
Doesn't have to adjust in this PR, but have we considered move all these kind of logic to python vs do them in Jinja

Reply:
I was thinking about this yesterday. Both `get_grant_sql` and `get_revoke_sql` now have the same boilerplate, which is unpacking the grant config and looping over them. The piece of this that really wants to be in Jinja is really more like:
```
{% macro get_grant_sql() %}
    grant {{ privilege }} on {{ relation }} to {{ grantee }}
{% endmacro %}
```
I could see refactoring to call this simpler macro instead, and move the unpacking/looping/array-building logic elsewhere.

For what it's worth, most of the heavier lifting is really happening in `apply_grants`. We could reimplemented some pieces of it in Python. Keeping the top-level macro call in Jinja (=user space) does allow people to override its behavior for themselves, though.

Reply:
I acted on this feedback in c763601, by moving the `grant_config` unpacking / looping logic into their own separate reusable macro, so that `get_grant_sql` + `get_revoke_sql` can be dead simple

Reply:
@jtcohen6 solid move! I also like @ChenyuLInx's notion of putting more looping logic into Python to make the Jinja simpler.